zookeeper/source/zk_nn_client.cc

```diff
@@ -342,6 +441,44 @@ namespace zkclient{
         return 0;
     }
 
+    bool ZkNnClient::blockDeleted(uint64_t uuid, std::string id) {
+        int error_code;
+        bool exists;
+
+        LOG(INFO) << "DataNode deleted a block with UUID " << std::to_string(uuid);
+
+        auto ops = std::vector<std::shared_ptr<ZooOp>>();
+
+        // Delete block locations
+        if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid) + "/" + id, exists, error_code)) {
+            if (exists) {
+                ops.push_back(zk->build_delete_op(BLOCK_LOCATIONS + std::to_string(uuid) + "/" + id));
+                // If deleted last child of block locations, delete block locations
+                std::vector <std::string> children = std::vector <std::string>();
+                if(!zk->get_children(BLOCK_LOCATIONS + std::to_string(uuid), children, error_code)){
+                    LOG(ERROR) << "getting children failed";
+                }
```

Comment:
If this DataNode is the only thing with the block, we should clean up BLOCK_LOCATIONS + std::to_string(uuid) too